lib/msal-common/src/unifiedCache/UnifiedCacheManager.ts

```diff
@@ -9,54 +9,65 @@ import { AccessTokenEntity } from "./entities/AccessTokenEntity";
 import { IdTokenEntity } from "./entities/IdTokenEntity";
 import { RefreshTokenEntity } from "./entities/RefreshTokenEntity";
 import { AccountEntity } from "./entities/AccountEntity";
-import { ICacheStorage } from "../cache/ICacheStorage";
 import { Deserializer } from "./serialize/Deserializer";
 import { Serializer } from "./serialize/Serializer";
 import { AccountCache } from "./utils/CacheTypes";
+import { ICacheStorageAsync } from "../cache/ICacheStorageAsync";
 
 export class UnifiedCacheManager {
 
     // Storage interface
     private inMemoryCache: InMemoryCache;
-    private cacheStorage: ICacheStorage;
+    private cacheStorage: ICacheStorageAsync;
 
-    constructor(cacheImpl: ICacheStorage) {
+    constructor(cacheImpl: ICacheStorageAsync) {
         this.cacheStorage = cacheImpl;
-        this.inMemoryCache = this.cacheStorage.getCache();
     }
 
     /**
      * setter  for in cache memory
      */
-    setCacheInMemory(cache: InMemoryCache): void {
+    async setCacheInMemory(cache: InMemoryCache): Promise<void> {
         this.inMemoryCache = cache;
     }
 
+    // if inMemmory cache is not set, set it from caheStorage
+    // this async operation can't be done in constructor
+    async setCacheInMemmoryIfNotSet(): Promise<void> {
```

Comment:
This is interesting and sorta unseltting.  The get cache function being async means that it cant be called in the constructor for the initial population. So in all the member functions I have had to add this set if not set function to make sure the internal state is created before any operations take place

Reply:
yes. This is the initial problem I had with async functions, so I wrapped async only for file access and made other APIs sync. I do see the value in this though, as in if we have (key, value) persistence, we should have a reasonable wrapper for async implementation. Let's brainstorm if there is a better approach.